internal/common/conversion/schema_generation.go

```diff
@@ -11,14 +11,44 @@ import (
 func DataSourceSchemaFromResource(rs schema.Schema, requiredFields []string, overridenFields map[string]dsschema.Attribute) dsschema.Schema {
 	blocks := convertBlocks(rs.Blocks, requiredFields)
 	attrs := convertAttrs(rs.Attributes, requiredFields)
-	for name, attr := range overridenFields {
-		if attr == nil {
-			delete(attrs, name)
-		} else {
-			attrs[name] = attr
+	overrideFields(attrs, overridenFields)
+	ds := dsschema.Schema{Attributes: attrs, Blocks: blocks}
+	UpdateSchemaDescription(&ds)
+	return ds
+}
+
+func PluralDataSourceSchemaFromResource(rs schema.Schema, requiredFields []string, overridenFields, overridenRootFields map[string]dsschema.Attribute, overrideResultsDoc string, hasLegacyFields bool) dsschema.Schema {
```

Comment:
Seems like a lot of arguments, can we use a Config struct instead?

Reply:
you're totally right, change here: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2844/commits/28ab07f8624eb4bc20c1b4dcf7129d7266d31245

also this will allow adding more params without changing all resources every time